alea/include/alps/alea/mpi.hpp

```diff
@@ -87,7 +87,7 @@ struct mpi_reducer
 
         // In-place requires special value for sendbuf, but only on root
         const void *sendbuf = am_root() ? MPI_IN_PLACE : data.data();
-        mpi::checked(MPI_Reduce(sendbuf, data.data(), data.size(), dtype_tag,
+        mpi::checked(MPI_Reduce(const_cast<void*>(sendbuf), data.data(), data.size(), dtype_tag,
```

Comment:
IMHO, it would be good to leave a comment that `const_cast` is needed here because of problems with MPI-2.1, but the cast is in fact safe. 

(Btw, if you'll make a commit adding the comment, please [add `[ci skip]`](https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build) to the commit message so that Travis won't try to test it again!)